ci/steps/attach_logs.py

```diff
@@ -1,179 +1,243 @@
+                s3_client=s3_client,
+                bucket_name=s3_bucket_name,
+                key=key,
+            )
+            name = os.path.basename(s3_log_key)
+            log_entry = glci.model.S3_ReleaseFile(
+                name=name,
+                suffix=name,
+                s3_key=s3_log_key,
+                s3_bucket_name=s3_bucket_name,
+            )
+            new_manifest = manifest.with_logfile(log_entry)
+            # upload manifest
+            manifest_path_suffix = manifest.canonical_release_manifest_key_suffix()
+            manifest_path = f'{glci.model.ReleaseManifest.manifest_key_prefix}/{manifest_path_suffix}'
+            glci.util.upload_release_manifest(
+              s3_client=s3_client,
+              bucket_name=s3_bucket_name,
+              key=manifest_path,
+              manifest=new_manifest,
+            )
+        else:
+            print(f'Task {task} was not built in this run')
 
 
 def _attach_and_upload_logs(
     architecture: str,
+    build_dict_str: str,
     build_targets: str,
     cicd_cfg_name: str,
     committish: str,
+    flavour_set: str,
     gardenlinux_epoch: str,
-    modifiers: str,
-    platform: str,
+    is_package_build: bool,
+    manifest_set_key: str,
+    platform_set: str,
+    promote_target: str,
     repo_dir: str,
     version: str,
-):
+) -> bool:
     cicd_cfg = glci.util.cicd_cfg(cfg_name=cicd_cfg_name)
     s3_client = glci.s3.s3_client(cicd_cfg)
     aws_cfg_name = cicd_cfg.build.aws_cfg_name
-    s3_bucket_name = cicd_cfg.build.s3_bucket_name
-    s3_package_bucket_name = cicd_cfg.package_build.s3_bucket_name
 
     log_path = os.path.join(repo_dir, 'build_log_full.zip')
     if not os.path.exists(log_path):
         print("No file found with log files, won't upload.")
         print("Exiting with failure, see logs from previous steps")
         return False
 
-    build_target_set = glci.model.BuildTarget.set_from_str(build_targets)
-    if not glci.model.BuildTarget.MANIFEST in build_target_set:
-        print(f'build target {glci.model.BuildTarget.MANIFEST=} not specified - skip upload')
+    prefix = datetime.datetime.utcnow().strftime('%Y%m%d-%H%M%S') + '-' + \
+        committish[:6] + '-'
+
+    if is_package_build:
+        s3_key = 'logs/' + prefix + 'package_build_log.zip'
+        s3_bucket_name = cicd_cfg.package_build.s3_bucket_name
+    else:
+        s3_key = 'objects/' + prefix + 'build_log.zip'
+        s3_bucket_name = cicd_cfg.build.s3_bucket_name
+
+    print(f'uploaded zipped logs to {s3_key=}')
+    uploaded_file = _upload_file(
+        log_file_path=log_path,
+        s3_key=s3_key,
+        s3_client=s3_client,
+        s3_bucket_name=s3_bucket_name
+    )
+
+    # write a file with the download URL so that it can be later added to the email
+    with open(os.path.join(repo_dir, 'log_url.txt'), 'w') as f:
+        f.write(f'https://gardenlinux.s3.eu-central-1.amazonaws.com/{s3_key}')
```

Comment:
at least pass-in `gardenlinux` (subdomain part of the url) as parameter read from current cfg